code/utils.py

```diff
@@ -7,6 +7,7 @@
 """
 
 
+from six.moves import xrange
```

Comment:
I'll change all the instances with range instead of xrange. That's what I did at first, but then I figured the proper way to use xrange in a compatible way. I'll just use range for all cases. They're "safe" in the sense that none of those calls require fast execution.